### PR TITLE
Highlight new log content, scroll immediately

### PR DESCRIPTION
**Pull request by daniel-beck**
Sometimes build logs are fairly repetitive, and it can be difficult to tell what's new from what was already there. If you've ever selected some text in the build log just to have a visual anchor to watch scroll up and out of view (or immediately disappear in particularly quickly scrolling build logs), you know what I mean.

This adds a small highlight animation to the block of log text that is newly added. Additionally, it replaces the old scroll behavior (that has an animation at least on my OS/browser) with one that explicitly requests instant scrolling.

https://user-images.githubusercontent.com/1831569/184113422-3d562336-56a0-4a09-8fdd-612bd72be403.mov

The current state isn't particularly pretty (as always, suggestions welcome!), but I think it has potential.

Right now, switching to a different tab, waiting multiple AJAX updates, then switching back to Jenkins will fire all animations of all newly added blocks, instead of only the last one. Unsure whether this makes sense or not (I tend to think it does, content is new to the viewer).

### Proposed changelog entries

* Add highlight to indicate newly added build log content.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6986"></a>

### CODE REVIEW

File: core/src/main/resources/lib/hudson/progressive-text.css [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+}
+
+@keyframes progressiveText_highlight {
+  0% {
+    background: rgba(0, 0, 100, 0.2);
+  }
+  100% {
+    background: rgba(0, 0, 100, 0);
+  }
+}
```
Comment (janfaracik) [inline core/src/main/resources/lib/hudson/progressive-text.css:12]: Looks good to me! Might be worth using a pseudo element like so: 

```
div.progressiveText-block:not(:first-child) {
  position: relative;

  &::before {
    content: "";
    position: absolute;
    inset: 0 -1rem;
    animation: progressiveText_highlight 1.5s ease;
    animation-fill-mode: both;
    background: var(--primary);
    opacity: 0.3;
    pointer-events: none;
  }
}

@keyframes progressiveText_highlight {
  to {
    opacity: 0;
  }
}
```

so we can use the --primary colour variable with opacity. This also allows us to have the flash background be full width.

Comment (daniel-beck) [reply core/src/main/resources/lib/hudson/progressive-text.css:12]: Could you provide that as CSS so I can simply insert it into the file and see what happens? I tried to "translate" but it doesn't actually disappear, it's just a colored background.

**Comment by jtnord** [comment]
Relates to https://issues.jenkins.io/browse/JENKINS-69587 ?

**Comment by daniel-beck** [comment]
> Relates to https://issues.jenkins.io/browse/JENKINS-69587 ?

Right, I noticed the bad behavior when scrolling described there and fixed it here for log scrolling specifically, because it was even worse with the additional color. I did not expect this was a change in Jenkins, as I had not seen any PR for it. Basil identified #6485 as the cause.

**Pull request closed without merging**